splipy/io/ofoam.py

```diff
@@ -0,0 +1,115 @@
+from itertools import groupby
+import numpy as np
+from operator import itemgetter
+from os.path import exists, isdir, join
+from os import makedirs
+
+from splipy import SplineModel
+
+
+class OpenFOAM(object):
+
+    def __init__(self, target):
+        self.target = target
+
+    def __enter__(self):
+        # Create the target directory if it does not exist
+        if not exists(self.target):
+            makedirs(target)
+        # If it does, ensure that it's a directory
+        elif not isdir(self.target):
+            raise FileExistsError('{} exists and is not a directory'.format(self.target))
+
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        pass
+
+    def _header(self, cls, obj, note=None):
+        s = 'FoamFile\n{\n'
+        s += '    version     2.0;\n'
+        s += '    format      ascii;\n'
+        s += '    class       %s;\n' % cls
+        if note:
+            s += '    note        "%s";\n' % note
+        s += '    object      %s;\n' % obj
+        s += '}\n'
+        return s
+
+    def write(self, model):
+        assert isinstance(model, SplineModel)
```

Comment:
This is in contrast to how the other `io` functions operate. They all take spline objects as input and output, not `SplineModel`. If you really need to keep it this way, then draw attention to it in some documentation and/or examples. If not, then take a look at `svg.py` on how models are written. Basically you need to have all splines read before starting to iterate (since bounding box needs to be on top), so the `read( )` call adds things to memory, while the `__exit__( )` does the actual writing. In the openFOAM case it would need to add all splines to a SplineModel object, compute topology and then write.

Reply:
I think we need to. We can add things to a SplineModel but the OpenFOAM converter can't know everything, like what the names of the boundaries need to be.

Reply:
Ah yes. Once you start to factor in all the meta-information such as names and boundary conditions and these things, then you are right.